### DISCUSSION TITLE
Another spotify issue

### DISCUSSION THREAD

**Discussion by boboschkah** [MiczFlor/RPi-Jukebox-RFID#2441] [Q&A]
Hi people!! First of all thanks for your incredible work! I am a complete noob in phoniebox! 

My phoniebox  is running on a pi 4b  4g i had laying around ! 

Fresh install of raspberry os lite legacy (bullseye) 

i have the recommended usb rfid reader 

so i followed the complete install guide like a good noob and everything went completely flawless

well except for spotify …

 i can log into iris and see my music i also put in my password again  

but it simply wont play music … not from a rfid swipe as well as directly in iris player !! 

Music in mp3 on the pi works all the way !!  

I couldnt find any problems in mupid config files .. i run it as a service … 

after hours and hours of searching here i finally came to the point of writing my very own very first q&a

**Comment by seschu** [comment]
Hi there

Does this maybe help you?
https://github.com/MiczFlor/RPi-Jukebox-RFID/issues/2429#issuecomment-2409105287

**Reply by boboschkah** [reply]
Just found that myself !! Actually i don't understand a word !  I don’t have a developer account for spotify!  I little desperate right now!

**Reply by seschu** [reply]
You just have to login with your spotify premium account.

**Reply by seschu** [reply]
How familiar are you with the shell in linux. I am just trying to find out how to adress your question. The devloper account is no big deal you can access developer.spotify.com with your regular account. After logging in you can press "Reveal your access token" on this websoite [developer.spotify.com](https://developer.spotify.com/documentation/web-playback-sdk/tutorials/getting-started)
You copy the string of letters and numbers to some editor where you can store it.

Then you create a file here at this location /var/lib/mopidy/spotify/credentials-cache/credentials.json

Note that you have to open the editor with sudo rights. (If you dont know how to do this just look for the words "linux sudo default editor temrinal how to"

This file should have this content
{"username":"weirdusername",
"auth_type":1,"auth_data":"generatedaccesstoken"}

generatedaccesstoken is the string of chaarcters you copied after pressing "Reveal your access token".
the "weirdusername" username is this username on spotify that I see on my profile edit page. It might be just your regular username on sotify. Just try what ever you have first:)

Then on the shell of the phonie box doenload this file https://github.com/kingosticks/gst-plugins-rs-build/releases/download/gst-plugin-spotify_0.14.0-alpha.1-1/gst-plugin-spotify_0.14.0.alpha.1-1_armhf.deb
With this command 
`wget https://github.com/kingosticks/gst-plugins-rs-build/releases/download/gst-plugin-spotify_0.14.0-alpha.1-1/gst-plugin-spotify_0.14.0.alpha.1-1_armhf.deb`
Then remove the old spotify streaming plugin 
`sudo apt remove gst-plugin-spotify`

and then install the downladed file in this way
 `sudo dpkg -i gst-plugin-spotify_0.14.0.alpha.1-1_armhf.deb`

Is this clearer? :)

**Comment by s-martin** [comment]
Spotify authentication has been fixed in #2435 

will be part of 2.8
